host/bloodview/src/data.c

```diff
@@ -59,13 +68,74 @@ static struct {
 	/** Acquisition channel to data channel. */
 	unsigned mapping[sizeof(unsigned) * CHAR_BIT];
 
+	/** Array of data channels. */
+	struct data_channel *channel;
+	unsigned channel_count; /**< Number of channels in channel array. */
+
 	/** Array of registered filters. */
 	struct data_filter *filter;
 
 	/** Number of entries in \ref filter. */
 	unsigned count;
 } data_g;
 
+/**
+ * Destroy channels.
+ */
+static void data__destroy_channels(void)
+{
+	if (data_g.channel != NULL) {
+		for (unsigned i = 0; i < data_g.channel_count; i++) {
+			fifo_destroy(data_g.channel[i].samples);
+		}
+		free(data_g.channel);
+		data_g.channel = NULL;
+	}
+
+	data_g.channel_count = 0;
+}
+
+/**
+ * Create channels.
+ *
+ * \param[in]  channel_mask  Acquition channel mask.
+ * \return true on success or false on error.
+ */
+static bool data__create_channels(unsigned channel_mask)
+{
+	unsigned channels = util_bit_count(channel_mask);
```

Comment:
It's not a big deal, but just FYI, there is a built-in function in GCC doing the same thing if you want to use. __builtin_popcount

Reply:
I tend to avoid compiler builtins unless there is any sort of performance requirement.

On the microcontroller I'd use the builtin!